Expensify/App
# Move ExistingOwners view into new component. Reuse confirm modal.

cc @aldo-expensify @jasperhuangg 

### Details
Just moving some things around so that our error handling is easier to follow.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/176349

### Tests
1. Follow steps here to get the modal to appear https://stackoverflow.com/c/expensify/questions/10740/10741
2. Verified this modal appears
![2021-09-02_13-18-33](https://user-images.githubusercontent.com/32969087/131927930-c8e5f740-cb87-4e16-8dac-3cc3310b9d40.png)
3. Pressed "Got it" and verified it went away
4. Make it appear again
5. Press the link and verify the modal closes and we navigate to the "Company Information" step.

### QA Steps
1. Create a policy and add User A and User B
1. Create bank account in open state with User A (via OldDot or via the `/bank-account` flow in NewDot)
1. Attempt to create bank account as User B in NewDot via the `/bank-account` flow
2. Verified this modal appears
3. Press "Got it" and verify it goes away
4. Make it appear again (add the same account)
5. Press the hyperlink and verify the modal closes and we navigate to the "Company Information" step.

**Note:** This cannot be QA'd on native platforms yet as the Plaid staging credentials do not work in NewDot yet.

### Tested On


### Screenshots


#### Web
![2021-09-07_12-07-41](https://user-images.githubusercontent.com/32969087/132416969-db4a42a7-6ce4-444f-8a6f-7f8ad7b2d151.png)

#### Mobile Web


#### Desktop


#### iOS


#### Android
![2021-09-07_12-09-11](https://user-images.githubusercontent.com/32969087/132417060-c8002a86-3bf3-447e-b11f-a1ec4e02811f.png)
